.github/workflows/cd.yml

```diff
@@ -2,10 +2,61 @@ name: CD
 
 on:
   push:
-    branches: [master]
+    branches:
+      - master
+      - prod
 
 jobs:
-  deploy:
+  api-deploy:
+    name: API Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          mask-aws-account-id: true
+
+      - id: login-ecr
+        name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Set branch-level configuration
+        run: |
+          if [[ ${{ github.ref }} == "refs/heads/prod" ]]; then
+            echo "IMAGE_TAG=prod" >> $GITHUB_ENV
+            echo "CLUSTER_NAME=${{ vars.AWS_ECS_API_CLUSTER_PROD }}" >> $GITHUB_ENV
+            echo "SERVICE_NAME=${{ vars.AWS_ECS_API_SERVICE_PROD }}" >> $GITHUB_ENV
+          else
+            echo "IMAGE_TAG=latest" >> $GITHUB_ENV
+            echo "CLUSTER_NAME=${{ vars.AWS_ECS_API_CLUSTER_STAGING }}" >> $GITHUB_ENV
+            echo "SERVICE_NAME=${{ vars.AWS_ECS_API_SERVICE_STAGING }}" >> $GITHUB_ENV
+          fi
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64
+
+      - name: Build, tag, and push docker image to Amazon ECR
+        env:
+          GIT_SHA: ${{ github.sha }}
+          IMAGE_NAME: ${{ steps.login-ecr.outputs.registry }}/${{ vars.AWS_ECR_API_REPO }}
+        run: |
+          docker build --platform linux/arm64 -t $IMAGE_NAME:$GIT_SHA -t $IMAGE_NAME:$IMAGE_TAG -f packages/daimo-api/Dockerfile .
+          docker push $IMAGE_NAME --all-tags
```

Comment:
do we want `--all-tags` or just stage/prod tag, depending on branch?

Reply:
With the current code we tag the SHA in addition to latest/prod (depending on the branch) and push both tags up.

<img width="473" alt="image" src="https://github.com/daimo-eth/daimo/assets/5856011/b284bfa6-8725-4fd6-a5ce-a4930c64dea2">

Would you prefer to just have the single latest/prod tag @dcposch?

Reply:
ah, i see.  that looks good

Reply:
as long as we're sure that a `master` build will never overwrite the `prod` tag

Reply:
The action's container will always  run in a fresh state so there won't be more than those two tags however I went ahead and updated the command to be more explicit for clarity.